vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
     "-class.-jw.util.version.Version-",
 }
 
+ECOSYSTEM_PKG_TYPE_MAPPING = {
+    "MAVEN": "maven",
+    "NUGET": "nuget",
+    "COMPOSER": "composer",
+    "PIP": "pypi",
+    "RUBYGEMS": "gem",
+    "GO": "golang",
+}
+
+ECOSYSTEM_VERSION_RANGE_MAPPING = {
+    "MAVEN": MavenVersionRange,
+    "NUGET": NugetVersionRange,
+    "COMPOSER": ComposerVersionRange,
+    "PIP": PypiVersionRange,
+    "RUBYGEMS": GemVersionRange,
+    "GO": GolangVersionRange,
+}
+
+PKG_TYPE_ECOSYSTEM_MAPPING = {value: key for (key, value) in ECOSYSTEM_PKG_TYPE_MAPPING.items()}
+
+QUERY = """
+        query{
+        securityVulnerabilities(first: 100, ecosystem: %s, %s) {
+            edges {
+            node {
+                advisory {
+                identifiers {
+                    type
+                    value
+                }
+                summary
+                references {
+                    url
+                }
+                severity
+                publishedAt
+                }
+                package {
+                name
+                }
+                vulnerableVersionRange
+            }
+            }
+            pageInfo {
+            hasNextPage
+            endCursor
+            }
+        }
+        }
+        """
+
 
 class GitHubTokenError(Exception):
     pass
 
 
+def get_item(json_object: dict, *attributes):
+    """
+    Return `item` by going through all the `attributes` present in the `json_object`
+    """
+    if not json_object:
+        LOGGER.error(f"json_object is empty: {json_object}")
+        return None
+    item = json_object
+    for attribute in attributes:
+        if not attribute in item:
+            LOGGER.error(f"Missing attribute {attribute} in {item}")
+            return None
+        item = item[attribute]
+    return item
+
+
 class GitHubAPIImporter(Importer):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    spdx_license_expression = "CC-BY-4.0"
+    token = os.environ["GH_TOKEN"]
+    endpoint = "https://api.github.com/graphql"
+
+    ecosystems = [
+        "MAVEN",
+        "NUGET",
+        "COMPOSER",
+        "PIP",
+        "RUBYGEMS",
+        "GO",
+    ]
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        headers = {"Authorization": f"token {self.token}"}
+        end_cursor_exp = ""
+        advisories = []
+        for ecosystem in self.ecosystems:
+            while True:
+                query_json = {"query": QUERY % (ecosystem, end_cursor_exp)}
```

Comment:
May be this is enough of a name?
```suggestion
                query = {"query": QUERY % (ecosystem, end_cursor_exp)}
```